athena-neptune/src/main/java/com/amazonaws/athena/connectors/neptune/NeptuneRecordHandler.java

```diff
@@ -0,0 +1,248 @@
+
+    /**
+     * used to aid in debugging. Athena will use this name in conjunction with your
+     * catalog id to correlate relevant query errors.
+     */
+    private static final String SOURCE_TYPE = "neptune";
+    private final NeptuneConnection neptuneConnection;
+    private GraphTraversalSource graphTraversalSource;
+
+    // private AmazonS3 amazonS3;
+
+    public NeptuneRecordHandler() {
+        this(AmazonS3ClientBuilder.defaultClient(), AWSSecretsManagerClientBuilder.defaultClient(),
+                AmazonAthenaClientBuilder.defaultClient(),
+                new NeptuneConnection(System.getenv("neptune_endpoint"), System.getenv("neptune_port")));
+    }
+
+    @VisibleForTesting
+    protected NeptuneRecordHandler(final AmazonS3 amazonS3, final AWSSecretsManager secretsManager,
+            final AmazonAthena amazonAthena, final NeptuneConnection neptuneConnection) {
+        super(amazonS3, secretsManager, amazonAthena, SOURCE_TYPE);
+        // this.amazonS3 = amazonS3;
+        this.neptuneConnection = neptuneConnection;
+        // this.graphTraversalSource = this.neptuneConnection.getTraversalSource();
+    }
+
+    /**
+     * Used to read the row data associated with the provided Split.
+     *
+     * @param spiller            A BlockSpiller that should be used to write the row
+     *                           data associated with this Split. The BlockSpiller
+     *                           automatically handles chunking the response,
+     *                           encrypting, and spilling to S3.
+     * @param recordsRequest     Details of the read request, including: 1. The
+     *                           Split 2. The Catalog, Database, and Table the read
+     *                           request is for. 3. The filtering predicate (if any)
+     *                           4. The columns required for projection.
+     * @param queryStatusChecker A QueryStatusChecker that you can use to stop doing
+     *                           work for a query that has already terminated
+     * @throws IOException
+     * @note Avoid writing >10 rows per-call to BlockSpiller.writeRow(...) because
+     *       this will limit the BlockSpiller's ability to control Block size. The
+     *       resulting increase in Block size may cause failures and reduced
+     *       performance.
+     */
+    @Override
+    protected void readWithConstraint(final BlockSpiller spiller, final ReadRecordsRequest recordsRequest,
+            final QueryStatusChecker queryStatusChecker) {
+        logger.info("readWithConstraint: enter - " + recordsRequest.getSplit());
+        final TableName tableName = recordsRequest.getTableName();
+        final String labelName = tableName.getTableName();
+        long numRows = 0;
+        final AtomicLong numResultRows = new AtomicLong(0);
+        final Client client = neptuneConnection.getNeptunClientConnection();
```

Comment:
I believe this client should be used in a try/catch/finally block so that we close it when done. Otherwise we might be leaking connections to Neptune. I also made a comment about using AutoClosable  on NeptuneConnection which could help.

Reply:
Need to investigate on AutoClosable, moved to try/catch/finally block